CONTRIBUTING.md

```diff
@@ -33,7 +44,7 @@ guidelines:
     ```
         git commit --amend
         [add the line, save and quit the editor]
-        git push -f
+        git push -f <repository> <branch>
```

Comment:
Specifying repository and branch should not be usually needed. Please drop this.

Reply:
Omitting repository and branch is only possible if the pull request branch was initially pushed to the fork using `git push --set-upstream <repo>`, or if the upstream branch has been set afterwards. Otherwise, the contributor will get the following error message about a missing upstream branch:

```
fatal: The current branch testo has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin test

To have this happen automatically for branches without a tracking
upstream, see 'push.autoSetupRemote' in 'git help config'.
```
A compromise to mark it as optional by placing it in square brackets


```suggestion
        git push -f [ <repository> <branch> ]
```

Reply:
Yeah adding `[]` works for me.

Reply:
thanks for your comments.

I've just pushed an update that uses the syntax from `man git-push`, which applies `[]` differently from Matthias' suggestion:  `git push -f [<repository> [<branch>]]`

Now that I look at it again, I'm not sure if it is better :-)

Let me know and I can change to `[<repository> <branch>]`

Reply:
I'm okay with both variants, and if that's the way you'll find it in the manual page, then why not use it?

Reply:
FWIW: I prefer `--force-with-lease` over `--force, -f` because it's safer when you push from multiple working copies. But that's another story.